presto-spark-base/src/main/java/com/facebook/presto/spark/execution/BatchTaskUpdateRequest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.presto.server.TaskUpdateRequest;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BatchTaskUpdateRequest
+{
+    private final TaskUpdateRequest taskUpdateRequest;
+    // Map from plan node id to serialized ShuffleReadInfo
+    private final Map<String, String> shuffleReadInfos;
```

Comment:
Should we make the shuffleReadInfos optional as well? (although we can have empty map to indicate that, but senmatic-wise, it'd be more consistent with the shuffleReadInfo)

Reply:
sure